### DISCUSSION TITLE
Is it possible to use multiple rtspclientsinks to send multiple AV sources to one rtsp-simple-server for publishing

### DISCUSSION THREAD

**Discussion by chaseology** [bluenviron/mediamtx#646] [Questions]
AV SOURCE SRC PAD... rtspclientsink location=rtsp://(ip address of the rtsp-simple-server):8554/mystream latency=0

the 1st Gstreamer pipeline RTSPCLIENTSINK works great and sends an audio source to the rtsp-simple-server...
i can hear that audio on the rtsp-simple-server  client receiver...

However,  the rtsp-simple-server seems to only allow one  rtspclientsink source to be sent to the rtsp-simple-server publish/INPUT...if i try a 2nd gstreamer pipeline as on top, above.. the rtsp-simple-server shuts down the 1st Gstreamer rtspclientsink pipeline, and also closes any of its receivers and then the receivers would have to be restarted...

**Comment by aler9** [comment]
Hello,

If you want to publish two separate streams, you have to use different path names:

```
AV SOURCE SRC PAD... rtspclientsink location=rtsp://(ip address of the rtsp-simple-server):8554/mystream1 latency=0
AV SOURCE SRC PAD... rtspclientsink location=rtsp://(ip address of the rtsp-simple-server):8554/mystream2 latency=0
```

Otherwise if you want to publish two tracks into a single path, you have to merge them with Gstreamer, create a single stream and send it once:

```
rtspclientsink name=s location=rtsp://localhost:8554/mystream source1src ! s.sink_0 source2src ! s.sink_1
```

**Reply by chaseology** [reply]
Thank you,  with the first method you provided, i was able to build this:
  https://youtu.be/bYSCj2EvMUQ

testing the 2nd method, using this script:
gst-launch-1.0 rtspclientsink name=s location=rtsp://localhost:8554/mystream  udpsrc port=2222 ! "audio/x-raw, layout=(string)interleaved, rate=(int)48000, format=(string)F32LE, channels=(int)2"  ! audioconvert !   s.sink_0  udpsrc port=3333 ! "audio/x-raw, layout=(string)interleaved, rate=(int)48000, format=(string)F32LE, channels=(int)2" ! audioconvert !  s.sink_1

it works, however the RTSP CLIENT pipeline audio outputs have a significant amount of low volume WHITE NOISE/(pink?) on each channel of each pipeline...even with ZERO audio input to the UDPsink pipeline FEEDs to these UDPsrc(s)...

any ideas/thoughts  on why there is a low level "noise" being produced ?
thank you

**Reply by chaseology** [reply]
Here is a picture of the noise on each of the 4 channels of the final RTSP CLIENT audio output - with ZERO AUDIO INPUT to the udpFEEDERs    
![noiseONpipes](https://user-images.githubusercontent.com/7384874/141516871-4bfd63b2-c006-4ca7-8f53-02ed4b1d9887.png)
s

**Reply by chaseology** [reply]
...after further study, it seems that "simple-rtsp-server" internal code is defaulting to  L8  (8 bit)  PCM  instead of 16 or 24 bit... is there a way to adjust the code to default to any of the higher bit rates in this use case as per the Gstreamer script as shown above:
 "audio/x-raw, layout=(string)interleaved, rate=(int)48000, format=(string)F32LE, channels=(int)2"
?
thank you...

**Reply by aler9** [reply]
Hello, the server internal code doesn't touch the track parameters, it's Gstreamer that is messing with sample depth. Try debugging the input and output pipelines to see where the degradation happens.

[a comment by an automated account is not shown]
